src/canmatrix/join.py

```diff
@@ -1,65 +1,72 @@
+# -*- coding: utf-8 -*-
+
+import typing
+
+import canmatrix.canmatrix as cm
```

Comment:
Due to `from canmatrix import *` killer in `__init__.py` I got unresolvable dependency loop when refactoring `copy` module #306 . It was simply not possible to `import canmatrix`. 

We must optimize the `__init__`, than I'm open to refactor to `import canmatrix`. With help of the IDE it will be super fast. Mypy has found more serious issues than this one...

Reply:
For me I don't need abbreviations.

any hints how to to the `__init__.py` right are welcome.
Is it as easy as move the content from `canmatrix.py` to `__init__.py` or is there a better/nicer way that we don't need to write `import canmatrix.canmatrix`?

Reply:
I'm not that skilled, but I already learned, that the problem of `from canmatrix.canmatrix import *` is, that we don't only import all classes (what we generally want), but also all module-level functions (what we usually don't want) and somehow also all modules, which the canmatrix.canmatrix imports. The python import system get confused and for example don't know, whether `import copy` should be the built-in `copy` or `canmatrix.copy`. And if two modules use `from another_modul import Something`, they generate dependency loop, which can't be resolved.

I'm not sure but I've impression that the `__init__` should rather contain concrete list, what to import, for example `from canmatrix.canmatrix import CanMatrix, Ecu, Frame, Signal, ...` so that it doesn't publish more, than needed. But again, I'm not sure.